nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/serialization/jaxb/JAXBSerializer.java

```diff
@@ -88,9 +98,15 @@ public T deserialize(final InputStream input) throws SerializationException {
         try {
             // Consume the header bytes.
             readDataModelVersion(input);
+
+            XMLStreamReader streamReader = XMLInputFactory.newInstance().createXMLStreamReader(input);
+
             final Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-            return (T) unmarshaller.unmarshal(input);
-        } catch (JAXBException e) {
+            JAXBElement<T> jaxbElement = unmarshaller.unmarshal(streamReader, clazz);
+            T deserializedObject = jaxbElement.getValue();
+
+            return deserializedObject;
```

Comment:
This could be collapsed into a single line.
```suggestion
            return jaxbElement.getValue();
```

Reply:
I find it tremendously useful - especially during debug - to have a single variable as a return value instead of an expression.

Reply:
I can see usefulness in a separate variable in some cases, particularly if it is the result of a complicated method. In this case, the value is a member variable of `JAXBElement`. IntelliJ flags this as redundant in the default configuration, which is not an absolute rule, but in this particular instance, having a separate variable does not seem to add much value for debugging.

Reply:
Again, I find it useful to return a single variable.
During debug, one can just put a breakpoint at the return statement and see right away what that value is.
Also it's great that one can actually give a descriptive _name_ for the variable which can help understand the code better.
All in all I consider this a matter of preference and plan to leave it as it is unless I see more suggestions form others to do otherwise.